### PR TITLE
Suggestions.

### PR DESCRIPTION
**Pull request by gmcode**
Possibly improve cross platform support.
Make validate work for C++11 because the library supports it.
Fix an unused variable mistake i made earlier.
Make constructor public, seems it should be. Possible compiler bug?

### CODE REVIEW

File: test/tz_test/validate.cpp [hunk 1]
```diff
@@ -128,10 +128,10 @@ tzmain()
             auto ymd = year_month_day(dp);
             auto time = make_time(begin - dp);
             std::cout << ymd << ' ' << time << "Z ";
-            if (info.offset >= 0s)
+            if (info.offset >= seconds(0))
                 std::cout << '+';
             std::cout << make_time(info.offset);
-            if (info.save == 0min)
+            if (info.save == minutes(0))
```
Comment (HowardHinnant) [inline test/tz_test/validate.cpp]: Stylistic nitpick:  How about we use `minutes{0}` (curlies) instead?

File: tz.cpp [hunk 2]
```diff
@@ -126,7 +126,7 @@ CONSTDATA auto max_day = date::dec/31;
 
 namespace detail
 {
-class undocumented {explicit undocumented() = default;};
+class undocumented {public: explicit undocumented() = default;};
```
Comment (HowardHinnant) [inline tz.cpp]: Wow, I don't know how that compiled.  Instead of putting public in there, lets change it to a `struct`.

File: tz.h [hunk 3]
```diff
@@ -1431,11 +1432,18 @@ parse(std::basic_istream<CharT, Traits>& is,
...
+#else
+                auto tt = timegm(&tm);
+#endif
+#if 0
 #ifndef _MSC_VER
                 auto tt = timegm(&tm);
 #else
                 auto tt = _mkgmtime(&tm);
 #endif
+#endif
```
Comment (HowardHinnant) [inline tz.h]: Ok, but remove the `if 0` part.

File: tz.cpp (hunk 2 above)
Comment (rhalbersma) [inline tz.cpp]: https://groups.google.com/a/isocpp.org/forum/?fromgroups#!topic/std-discussion/ShmXFZg-ikQ

serendipity?

**Comment by HowardHinnant** [comment]
Thanks, merged.

**Pull request closed without merging**